demo-transactions/src/main/kotlin/io/holixon/cqrshexagonaldemo/demoparent/transactions/adapter/outbound/account/jpa/AccountOutAdapter.kt

```diff
@@ -26,4 +28,40 @@ class AccountOutAdapter(
         val savedAccount = jpaAccountOutAdapter.save(toEntity)
         return accountEntityMapper.toDomain(savedAccount)
     }
-}
\ No newline at end of file
+
+    //Refactor code similar to withdraw...
+    @Transactional
+    override fun deposit(iban: Iban, amount: BigDecimal): Account {
```

Comment:
This is not according to hexagonal architecture. Here the AccountOutAdapter contains the logic what methods to call on the account domain object. Correct would be to just return the Account via some findById(), modify it in the useCase (call account.deposit) and then save it again in the OutAdapter.

Also I'm not sure where to put the @Transactional then. It should be on the InAdapter I think.

Reply:
You are right! im mixing domain class logic in the adapter breaking the fundamental rule 😅  Since this branch is just to add the balance to the account entity i will remove this deposit and withdraw logic and fix it on the next branch that is the implementation of it and im working on it also.